pkg/controller/utils/datadog/metrics_forwarder.go

```diff
@@ -489,31 +488,38 @@ func (mf *metricsForwarder) getDatadogAgent() (*datadoghqv1alpha1.DatadogAgent,
 }
 
 // getCredentials returns the Datadog API Key and APP Key, it returns an error if one key is missing
+// getCredentials tries to get the credentials from the CRD, then from operator configuration
 func (mf *metricsForwarder) getCredentials(dda *datadoghqv1alpha1.DatadogAgent) (string, string, error) {
+	apiKey, appKey, err := mf.getCredsFromDatadogAgent(dda)
+	if err != nil {
+		if errors.Is(err, ErrEmptyAPIKey) || errors.Is(err, ErrEmptyAPPKey) {
```

Comment:
This means that we forbid a behaviour where the API key could be retrieved from the `datadog` CR and only the APP key fetch is fallen back to the operator config.
As soon as one of the two keys cannot be retrieved from the CR, the two are fetched from the operator config even if the other key in the CR is valid.

I don’t know if there a real need for such a setup, but a user who defines only an API key in its CRD might expect to have the API key of the CRD used and to have only the APP key fallen back to the operator one.

At least, this differs from the previous behaviour.

Reply:
TBH I did this on purpose, I'm thinking it's more natural to retrieve the app and api keys from the same place either the CRD or the operator config. Especially with the Datadog Monitor controller where the only source is the operator config.

Reply:
In this case, should we put a log to warn the user if one of the two keys is set in the `datadog` CR but is not used because the other one isn’t present ?

It’s just that a user might expect the CR specific settings to override the generic operator ones.